include/aws/s3/private/s3_request.h

```diff
@@ -126,6 +126,9 @@ struct aws_s3_request {
      */
     uint32_t part_number;
 
+    /* The upload_timeout used. Zero, if the request is not a upload part */
+    size_t upload_timeout_ms;
```

Comment:
general feedback: stop using size_t by default. It's good for the number of items in memory, but it's weird as milliseconds, or total number of requests sent